pkg/testutils/moduletemplate.go

```diff
@@ -27,7 +36,7 @@ func GetModuleTemplate(ctx context.Context,
 	}
 
 	moduleReleaseMeta, err := GetModuleReleaseMeta(ctx, module.Name, namespace, clnt)
-	if client.IgnoreNotFound(err) != nil {
+	if !meta.IsNoMatchError(err) && client.IgnoreNotFound(err) != nil {
 		return nil, fmt.Errorf("failed to get ModuleReleaseMeta: %w", err)
 	}
```

Comment:
to reviewer: this is what caused the trouble. When the CRD was not synced to the SKR cluster, this returned and error. In turn, the retries of creating the ModuleTemplates were failing because of the set `resourceVersion` and we ended in an endless loop.
It worked out when the other parallel tests including ModuleReleaseMeta had synced the CRD to the SKR beforehand.

Reply:
Can we use this reset of the `resourceVersion` during the tearDown of the tests. that the next test suite will start fresh? or it's just for this specific case?